OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -253,5 +246,159 @@ public void SaveVariation(Experiment experiment, Variation variation, UserProfil
                 ErrorHandler.HandleError(new Exceptions.OptimizelyRuntimeException(exception.Message));
             }
         }
+
+        /// <summary>
+        /// Try to bucket the user into a rollout rule.
+        /// Evaluate the user for rules in priority order by seeing if the user satisfies the audience.
+        /// Fall back onto the everyone else rule if the user is ever excluded from a rule due to traffic allocation.
+        /// </summary>
+        /// <param name = "featureFlag" >The feature flag the user wants to access.</param>
+        /// <param name = "userId" >User Identifier</param>
+        /// <param name = "filteredAttributes" >The user's attributes. This should be filtered to just attributes in the Datafile.</param>
+        /// <returns>null if the user is not bucketed into the rollout or if the feature flag was not attached to a rollout.
+        /// otherwise the Variation the user is bucketed into</returns>
+        public virtual Variation GetVariationForFeatureRollout(FeatureFlag featureFlag, string userId, UserAttributes filteredAttributes)
+        {
+            if (string.IsNullOrEmpty(featureFlag.RolloutId))
+            {
+                Logger.Log(LogLevel.INFO, $"The feature flag \"{featureFlag.Key}\" is not used in a rollout.");
+                return null;
+            }
+
+            Rollout rollout = ProjectConfig.GetRolloutFromId(featureFlag.RolloutId);
+
+            if (string.IsNullOrEmpty(rollout.Id))
+            {
+                Logger.Log(LogLevel.ERROR, $"The rollout with id \"{featureFlag.RolloutId}\" is not found in the datafile for feature flag \"{featureFlag.Key}\"");
+                return null;
+            }
+
+            Variation variation = null;
+            var rolloutRulesLength = rollout.Experiments.Count;
+
+            if (rolloutRulesLength == 0)
+                return null;
```

Comment:
Can we log a message saying there aren't any targeting rules in the rollout? I think we'll need to add this to the other SDK implementations too (Ruby)

Reply:
we will always have the everyone else rule.

Reply:
Okay, I got it. The check and relevant unit test has been removed.